base/base.bzl

```diff
@@ -48,79 +49,107 @@ def distro_components(distro_suffix):
-        env = {"PATH": "$$PATH:/busybox"},
-        tars = ["//experimental/busybox:busybox.tar"],
-    )
+            # A debug image with busybox available.
+            container_image(
+                name = "debug_" + user + "_" + arch + distro_suffix,
+                architecture = arch,
+                base = ":base_" + user + "_" + arch + distro_suffix,
+                directory = "/",
+                entrypoint = ["/busybox/sh"],
+                env = {"PATH": "$$PATH:/busybox"},
+                tars = ["//experimental/busybox:busybox_" + arch + ".tar"],
+            )
 
-    container_image(
-        name = "base_nonroot" + distro_suffix,
-        base = ":base" + distro_suffix,
-        user = "%d" % NONROOT,
-        workdir = "/home/nonroot",
-    )
+        ##########################################################################################
+        # Check that we can overlay a pure Go binary on a static base to check certificates
+        ##########################################################################################
+        go_binary(
+            name = "check_certs_" + arch + distro_suffix,
+            srcs = ["testdata/check_certs.go"],
+            goarch = arch,
+            # Test image is linux based
+            goos = "linux",
+            pure = "on",
+        )
 
-    container_image(
-        name = "debug_nonroot" + distro_suffix,
-        base = ":debug" + distro_suffix,
-        user = "%d" % NONROOT,
-        workdir = "/home/nonroot",
-    )
+        container_image(
+            name = "check_certs_image_" + arch + distro_suffix,
+            base = "//base:static_root_" + arch + distro_suffix,
+            files = [":check_certs_" + arch + distro_suffix],
+            symlinks = {
+                "/check_certs": "check_certs_" + arch + distro_suffix,
+            },
+            visibility = ["//visibility:private"],
+        )
 
-    container_test(
-        name = "debug" + distro_suffix + "_test",
-        configs = ["testdata/debug.yaml"],
-        image = ":debug" + distro_suffix,
-    )
+        container_test(
+            name = "static_" + arch + distro_suffix + "_test",
+            configs = ["testdata/static.yaml"],
+            image = ":check_certs_image_" + arch + distro_suffix,
+            tags = ["manual", arch],
+        )
 
-    container_test(
-        name = "base" + distro_suffix + "_test",
-        configs = ["testdata/base.yaml"],
-        image = ":base" + distro_suffix,
-    )
+        ##########################################################################################
+        # Check that we can invoke openssl in the base image to check certificates.
+        ##########################################################################################
+        container_test(
+            name = "openssl_" + arch + distro_suffix + "_test",
+            configs = ["testdata/certs.yaml"],
+            image = ":base_root_" + arch + distro_suffix,
```

Comment:
Is it meaningful to parameterize user (here and tests below) given that we're in a loop, or it is not worth?

Reply:
We could look at doing this more, it wasn't clear to me if/which of these tests might do something privileged, so I tried to avoid creeping the scope too much.

The one thing I did make a point to do was make the pure-go test use static, which I believe previously had zero direct test coverage.